kernel/src/platform/platform.rs

```diff
@@ -31,6 +32,12 @@ pub trait KernelResources<C: Chip> {
     /// the kernel will use.
     type ContextSwitchCallback: ContextSwitchCallback;
 
+    /// Implements the kernel policy for checking cryptographic credentials in
+    /// TBF objects to determine if the contained application can be run as a
+    /// process. If `None` then no credentials are checked and the kernel runs
+    /// all successfully loaded application binaries.
+    type Verifier: process_checking::AppVerifier<'static>;
```

Comment:
note: this adds a `'static` requirement that I don't believe was there before, which can be more problematic to test